bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeCameraHandler.java

```diff
@@ -12,106 +12,44 @@
+        switch (channelId) {
             case CHANNEL_WELCOME_CAMERA_STATUS:
-                return module != null ? toOnOffType(module.getStatus()) : UnDefType.UNDEF;
+                return getStatusState();
             case CHANNEL_WELCOME_CAMERA_SDSTATUS:
-                return module != null ? toOnOffType(module.getSdStatus()) : UnDefType.UNDEF;
+                return getSdStatusState();
             case CHANNEL_WELCOME_CAMERA_ALIMSTATUS:
-                return module != null ? toOnOffType(module.getAlimStatus()) : UnDefType.UNDEF;
+                return getAlimStatusState();
             case CHANNEL_WELCOME_CAMERA_ISLOCAL:
-                return (module == null || module.getIsLocal() == null) ? UnDefType.UNDEF
-                        : module.getIsLocal() ? OnOffType.ON : OnOffType.OFF;
+                return getIsLocalState();
             case CHANNEL_WELCOME_CAMERA_LIVEPICTURE_URL:
-                return getLivePictureURL() == null ? UnDefType.UNDEF : toStringType(getLivePictureURL());
+                return getLivePictureURLState();
             case CHANNEL_WELCOME_CAMERA_LIVEPICTURE:
-                return getLivePictureURL() == null ? UnDefType.UNDEF : HttpUtil.downloadImage(getLivePictureURL());
+                return getLivePictureState();
             case CHANNEL_WELCOME_CAMERA_LIVESTREAM_URL:
-                return getLiveStreamURL() == null ? UnDefType.UNDEF : new StringType(getLiveStreamURL());
-        }
-        return super.getNAThingProperty(chanelId);
-    }
-
-    /**
-     * Get the url for the live snapshot
-     *
-     * @return Url of the live snapshot
-     */
-    private String getLivePictureURL() {
-        String result = getVpnUrl();
-        if (result != null) {
-            result += LIVE_PICTURE;
-        }
-        return result;
-    }
-
-    /**
-     * Get the url for the live stream depending wether local or not
-     *
-     * @return Url of the live stream
-     */
-    private String getLiveStreamURL() {
-        String result = getVpnUrl();
-        if (result != null) {
-            result += "/live/index";
-            result += isLocal() ? "_local" : "";
-            result += ".m3u8";
+                return getLiveStreamState();
         }
-        return result;
-    }
-
-    @SuppressWarnings("null")
-    private String getVpnUrl() {
-        return (module == null) ? null : module.getVpnUrl();
-    }
-
-    public String getStreamURL(String videoId) {
-        String result = getVpnUrl();
-        if (result != null) {
-            result += "/vod/" + videoId + "/index";
-            result += isLocal() ? "_local" : "";
-            result += ".m3u8";
-        }
-        return result;
-    }
-
-    @SuppressWarnings("null")
-    private boolean isLocal() {
-        return (module == null || module.getIsLocal() == null) ? false : module.getIsLocal().booleanValue();
+        return super.getNAThingProperty(channelId);
     }
-}
+}
```

Comment:
Some of our build tools don't like EOF without a newline. So add a newline here.

Reply:
done